### PR TITLE
saasherder validate promotion of hotfix versions

### PR DESCRIPTION
**Pull request by maorfr**
part of https://issues.redhat.com/browse/APPSRE-10364

depends on https://github.com/app-sre/qontract-schemas/pull/669

with this PR, commit sha which will be defined under `hotfixVersions` for a `codeComponent` will be able to be promoted even being published with success to any subscribed channels (automated promotion).

### CODE REVIEW

File: reconcile/utils/saasherder/interfaces.py [hunk 1]
```diff
@@ -356,6 +359,11 @@ class SaasServiceOwner(Protocol):
     email: str
 
 
+class AppCodeComponent(Protocol):
+    url: str
+    hotfix_versions: list[str]
```
Comment (chassing) [inline reconcile/utils/saasherder/interfaces.py:364]:
```suggestion
    hotfix_versions: Optional[list[str]]
```

Comment (maorfr) [reply reconcile/utils/saasherder/interfaces.py]: AppCodeComponent(Protocol):

File: reconcile/utils/saasherder/saasherder.py [hunk 2]
```diff
@@ -1889,6 +1899,9 @@ def _validate_promotion(self, promotion: Promotion) -> bool:
         if not promotion.subscribe:
             return True
 
+        if promotion.commit_sha in self.hotfix_versions.get(promotion.url, set()):
```
Comment (fishi0x01) [inline reconcile/utils/saasherder/saasherder.py:1902]: nit: can we add a small comment or dedicated func to highlight that this is a general gate switch, i.e., it must always run before any other gate is evaluated? Im afraid I might accidentally miss that in the future.

**Review by fishi0x01 — APPROVED** [review]
LGTM :rocket:

File: reconcile/utils/saasherder/saasherder.py (hunk 2 above)
Comment (maorfr) [reply reconcile/utils/saasherder/saasherder.py:1903]: https://github.com/app-sre/qontract-reconcile/pull/4434/commits/6f6aef7c3981af1d3c8b8d1ed335500d06ae8f42

**Pull request merged**